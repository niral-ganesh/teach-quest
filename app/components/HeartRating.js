import React from "react";
import { View, StyleSheet } from "react-native";
import { FontAwesome } from "@expo/vector-icons"; // Import FontAwesome from Expo
import Colors from "../config/Colors";

const HeartRating = ({ rating }) => {
  // Create an array with 5 elements for 5 hearts
  const hearts = Array(5).fill(0);

  return (
    <View style={styles.container}>
      {hearts.map((_, index) => (
        <FontAwesome
          key={index}
          name={index < rating ? "heart" : "heart-o"} // Full heart if index is less than rating, otherwise empty heart
          size={16}
          color={Colors.red}
          style={styles.heart}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
  },
  heart: {
    marginHorizontal: 2,
  },
});

export default HeartRating;
