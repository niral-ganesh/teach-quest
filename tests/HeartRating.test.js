// HeartRating.test.js
import React from "react";
import { render } from "@testing-library/react-native";
import HeartRating from "../app/components/HeartRating";

it("renders correctly with rating 5", () => {
  const rating = 5;
  const tree = render(<HeartRating rating={rating} />);
  expect(tree).toMatchSnapshot();
});
