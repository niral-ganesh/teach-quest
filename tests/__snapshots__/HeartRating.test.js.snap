// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with rating 5 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <Text />
  <Text />
  <Text />
  <Text />
  <Text />
</View>
`;
